include/zephyr/dt-bindings/pinctrl/silabs-pinctrl-siwx917.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_ZEPHYR_DT_BINDINGS_PINCTRL_SILABS_PINCTRL_SIWX917_H_
+#define INCLUDE_ZEPHYR_DT_BINDINGS_PINCTRL_SILABS_PINCTRL_SIWX917_H_
+
+#include <zephyr/dt-bindings/dt-util.h>
+
+#if !defined(FIELD_PREP)
+/* Upstream does not make these macros available to DeviceTree */
+#define LSB_GET(value)          ((value) & -(value))
+#define FIELD_GET(mask, value)  (((value) & (mask)) / LSB_GET(mask))
+#define FIELD_PREP(mask, value) (((value) * LSB_GET(mask)) & (mask))
+#endif
+
+#define SIWX917_PINCTRL_PORT_MASK    0x0000000FUL
+#define SIWX917_PINCTRL_PIN_MASK     0x000000F0UL
+#define SIWX917_PINCTRL_ULPPIN_MASK  0x00000F00UL
+#define SIWX917_PINCTRL_MODE_MASK    0x0003F000UL
+#define SIWX917_PINCTRL_ULPMODE_MASK 0x00FC0000UL
+#define SIWX917_PINCTRL_PAD_MASK     0xFF000000UL
+
+/* Declare an integer representing the pinctrl/gpio state of a signal on SiWx917.
+ * @param mode          HP GPIO mode, 0xFF if HP GPIO mux isn't used
+ * @param ulpmode       ULP GPIO mode, 0xFF if ULP GPIO mux isn't used
+ * @param pad           HP pad number, 0xFF if HP pad isn't used, 0 if host pad
+ * @param port          GPIO port number (0-4)
+ * @param pin           HP GPIO pin number, 0 if not used
+ * @param ulppin        ULP GPIO pin number, 0 if not used
+ */
```

Comment:
This comment does not help very much to understand how to fill the parameters. 
BTW, I believe there is no public ressource describing how these values interact.

I have to admit I don't know how to provide a useful documentation to the user.

Otherwise, are you sure `ulppin == 0` means it is not used?

Reply:
Users don't need to care, they need to use the macros that make use of this builder macro. Developers may need to care, but they need to understand the 6 concepts anyway. I'll see if I can come up with a short description though.

Both instances of `0 if not used` here is really `don't care if corresponding mode isn't set`.

Reply:
I believe in this line, `ulppin == 0` is not ignored:
```
#define UART2_RTS_ULP0      SIWX917_GPIO( 9,    6, 22, 4,  0,  0)
```

Reply:
Yeah, the comment was meant as "just set it to 0 if you aren't using it", not as "if it is zero, it isn't used". A good reason to change it from `0` to `don't care`.